bazel/PPROF.md

```diff
@@ -0,0 +1,131 @@
+# Memory consumption testing with `pprof`
+
+To use `pprof` to analyze performance and memory consumption in Envoy, you can
+use the built-in statically linked profiler, or dynamically link it in to a
+specific place yourself.
+
+# Linking
+
+## Static Linking
+
+Static linking is already available (because of a `HeapProfilerDump()` call
+inside
+[`Envoy::Profiler::Heap::forceLink()`](https://github.com/envoyproxy/envoy/blob/master/source/common/profiler/profiler.cc#L21-L26)).
+
+### Compiling a statically-linked Envoy
+
+Just compile Envoy as normal:
+
+    $ bazel build //source/exe:envoy-static
+
+### Running a statically-linked Envoy with `pprof`
+
+And run the binary with a `HEAPPROFILE` env var, like so:
```

Comment:
nit: remove `And`.

Also, can you add a brief explanation of what the `HEAPPROFILE` environment variable does (It's explained below, so just saying it will be explained in section X would work too)?